File: algorithms/2_medium/F/01985_find_the_kth_largest_integer_in_the_array/sort.cpp
class Solution {
public:
    static string kthLargestNumber(vector<string>& nums, const int k) {
        const ptrdiff_t len = std::ssize(nums);
        assert(k >= 1 && k <= len);
        ranges::sort(nums, [](const string& a, const string& b) {
            if (a.size() != b.size()) {
                return a.size() < b.size();
            } else {
                return a < b;
            }
        });

        return nums[len - k];
    }
};

File: algorithms/2_medium/F/01985_find_the_kth_largest_integer_in_the_array/sort.rs
impl Solution {
    pub fn kth_largest_number(mut nums: Vec<String>, k: i32) -> String {
        let len = nums.len();
        assert!(k >= 1 && (k as usize) <= len);
        nums.sort_unstable_by(|a, b| {
            if a.len() != b.len() {
                a.len().cmp(&b.len())
            } else {
                a.cmp(&b)
            }
        });

        nums.swap_remove(len - (k as usize))
    }
}
